### PR TITLE
Make Vulnerability objects comparable by value

### PR DESCRIPTION
**Pull request by SaberStrat**
Enables comparison of two Vulnerability objects by their fields' values.

Implements #463

### CODE REVIEW

**Comment by nscuro** [comment]
@SaberStrat FYI the build failures should be fixed by #469, I am currently waiting for a review on that. Once merged, you can rebase your branch and the build will work.

**Comment by VinodAnandan** [comment]
@SaberStrat thank you for your PR. The https://github.com/CycloneDX/cyclonedx-core-java/pull/469 PR has been merged. Could you please rebase your PR?

**Comment by SaberStrat** [comment]
Rebased for #469

**Review by nscuro — APPROVED** [review]

**Pull request merged**
